access-modifier-annotation/src/main/java/org/kohsuke/accmod/restrictions/Beta.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2018 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.kohsuke.accmod.restrictions;
+
+import org.kohsuke.accmod.impl.ErrorListener;
+import org.kohsuke.accmod.impl.Location;
+import org.kohsuke.accmod.impl.RestrictedElement;
+
+/**
+ * References are only allowed within the same module, as in {@link NoExternalUse},
+ * or when a special flag is set in the consuming module.
+ * This is the property {@code useBeta} with the value {@code true}.
+ */
+public class Beta extends DoNotUse {
```

Comment:
Would be great to add support of some info/justification text to `DoNotUse ` so that error/warning messages can provide more details. Not a part for this PR I'd guess

Reply:
I think you can just cover that in Javadoc.